esp/esp/program/modules/handlers/studentclassregmodule.py

```diff
@@ -479,7 +492,7 @@ def catalog_render(self, request, tl, one, two, module, extra, prog, timeslot=No
 
         class_category_id = None
         for cls in classes:
-            if cls.category.id != class_category_id:
+            if cls.category.id != class_category_id and categories_sort:
```

Comment:
oh man does this finally fix the old bug where if you don't sort by category you get a bazillion section headings!?

Reply:
Yes, finally!